From 43356ea8c42c411fafbadcebc75b9db90b0a68c8 Mon Sep 17 00:00:00 2001
From: Sanghee Kim <sh0130.kim@samsung.com>
Date: Thu, 5 Sep 2019 12:25:51 +0900
Subject: [PATCH] mbedtls: esp: disable not used features

For binary size optimization.

MBEDTLS_PK_PARSE_EC_EXTENDED : 896 bytes
MBEDTLS_GENPRIME             : 1760 bytes
MBEDTLS_PKCS1_V21            : 2720 bytes
MBEDTLS_MD5_C                : 4656 bytes
MBEDTLS_SHA1_C               : 5760 bytes

Total                        : 15792 bytes

Signed-off-by: Sanghee Kim <sh0130.kim@samsung.com>
---
 components/mbedtls/port/include/mbedtls/esp_config.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/components/mbedtls/port/include/mbedtls/esp_config.h b/components/mbedtls/port/include/mbedtls/esp_config.h
index c8f682e..ad82896 100644
--- a/components/mbedtls/port/include/mbedtls/esp_config.h
+++ b/components/mbedtls/port/include/mbedtls/esp_config.h
@@ -630,7 +630,7 @@
  *
  * Disable if you only need to support RFC 5915 + 5480 key formats.
  */
-#define MBEDTLS_PK_PARSE_EC_EXTENDED
+//#define MBEDTLS_PK_PARSE_EC_EXTENDED

 /**
  * \def MBEDTLS_ERROR_STRERROR_DUMMY
@@ -654,7 +654,7 @@
  *
  * Requires: MBEDTLS_BIGNUM_C
  */
-#define MBEDTLS_GENPRIME
+//#define MBEDTLS_GENPRIME

 /**
  * \def MBEDTLS_FS_IO
@@ -703,7 +703,7 @@
  *
  * This enables support for RSAES-OAEP and RSASSA-PSS operations.
  */
-#define MBEDTLS_PKCS1_V21
+//#define MBEDTLS_PKCS1_V21

 /**
  * \def MBEDTLS_SELF_TEST
@@ -1648,7 +1648,7 @@
  * This module is required for SSL/TLS and X.509.
  * PEM_PARSE uses MD5 for decrypting encrypted keys.
  */
-#define MBEDTLS_MD5_C
+//#define MBEDTLS_MD5_C

 /**
  * \def MBEDTLS_NET_C
-- 
2.7.4

